### ISSUE TITLE
Add class to run a collection of transformers for data processing

### ISSUE THREAD

**Issue by RNKuhns** [alan-turing-institute/sktime#874]
**Is your feature request related to a problem? Please describe.**
During exploratory data analysis it would be nice to be able to have an easy way to run several transformations (say a box-cox transformation and then a detrender) in the specified order in order to re-use the transformed series for different exploratory data analysis tasks. 

**Describe the solution you'd like**
 Sktime has functionality similar to this built into `TransformedTargetForecaster`.  @aiwalter also has this functionality built-into his proposed `ForecastingPipeline` (See #844) that also includes the ability to do transformations to exogenous data (not just the endogenous series). 

An initial solution would be to just handle transformation of the endogenous (Y) data, ala `TransformedTargetForecaster`. 

In that initial case, I'd propose the transformer functionality of `TransformedTargetForecaster` be added in a separate class called `TransformerPipeline`.  While this doesn't seem like it adds a lot of functionality, the benefit would be making it more streamlined for users to do exploratory data analysis and avoiding the need for users to recreate code that sktime is implicitly using internally. 

For example, in the simple case that ignores exogenous data:

```python
processing_pipe = TransformerPipeline(
    [
        ('boxcox', BoxCoxTransformer(...)), 
        ('detrend', DeTrender(...)), 
    ]
)
yt = processing_pipe.fit_transform(y)
plot_series(yt)
```
A more complete solution would be to build this into the framework in #844 or 
Or in the case that parallels #844: 
```python
processing_pipe = TransformerPipeline(
    [
        ('boxcox', BoxCoxTransformer(...), ["y"]), 
        ('detrend', DeTrender(...), ["y"]), 
        ("scaler", TabularToSeriesAdaptor(MinMaxScaler()), ["y", "X"]),
    ]
)
yt, xt = processing_pipe.fit_transform(y, x)
# The function below is hypothetical but would be a potential use-case if it existed
plot_cross_correlation(yt, xt)
```

**Describe alternatives you've considered**
The primary alternative is for users to manually loop through the desired transformations and apply them, but having a concise way to do this would make it easy for users to get right to exploratory data analysis.  

Plus, by explicitly having a `TransformerPipeline` we'd be exposing the functionality that sktime is already using internally, so users don't have to write the code on their own.

**Comment by aiwalter** [comment]
@RNKuhns lets integrate this into `TransformedTargetForecaster`? I will then also have it in the future `ForecastingPipeline`..

**Comment by aiwalter** [comment]
I just created a new issue for `TransformedTargetForecaster` to do what you wrote here. Maybe lets close this here then? @RNKuhns 
#1090

**Comment by RNKuhns** [comment]
Closed in favor of #1090.
